velox/dwio/common/Options.h

```diff
@@ -605,18 +605,13 @@ struct WriterOptions {
   /// can choose to implement its custom memory reclaimer if needed and not use
   /// this default one.
   std::function<std::unique_ptr<velox::memory::MemoryReclaimer>()>
-      defaultMemoryReclaimerFactory{[]() { return nullptr; }};
+      memoryReclaimerFactory{[]() { return nullptr; }};
 
   std::optional<velox::common::CompressionKind> compressionKind;
-  std::optional<uint64_t> orcMinCompressionSize{std::nullopt};
```

Comment:
I would say we prefer to keep compile time key instead of runtime key at file option level.  You can move the DWRF/ORC specific ones to `dwrf::WriterOptions`